File: solutions/elixir/resistor-color-duo/1/lib/resistor_color_duo.ex
defmodule ResistorColorDuo do
  @doc """
  Calculate a resistance value from two colors
  """
  @spec value(colors :: [atom]) :: integer
  def value(colors) do
    [first, second | _] = Enum.take(colors, 2)
    List.to_integer(~c"#{get_color_val(first)}#{get_color_val(second)}")
  end

  defp get_color_val(color) do
    case color do
      :black -> 0
      :brown -> 1
      :red -> 2
      :orange -> 3
      :yellow -> 4
      :green -> 5
      :blue -> 6
      :violet -> 7
      :grey -> 8
      :white -> 9
    end
  end
  
end

File: solutions/c/resistor-color-duo/1/resistor_color_duo.c
#include "resistor_color_duo.h"

int map_color_to_int(resistor_band_t color) {
  switch (color) {
    case BLACK:
      return 0;
      break;
    case BROWN:
      return 1;
      break;
    case RED:
      return 2;
      break;
    case ORANGE:
      return 3;
      break;
    case YELLOW:
      return 4;
      break;
    case GREEN:
      return 5;
      break;
    case BLUE:
      return 6;
      break;
    case VIOLET:
      return 7;
      break;
    case GREY:
      return 8;
      break;
    case WHITE:
      return 9;
      break;
  }
}

int color_code(resistor_band_t *pair) {
  resistor_band_t colour_1 = pair[0];
  resistor_band_t colour_2 = pair[1];
  int colour_1_int = map_color_to_int(colour_1);
  int colour_2_int = map_color_to_int(colour_2);
  int result = (10 * colour_1_int) + colour_2_int;
  return result;
}
